interoperator/controllers/multiclusterdeploy/provisioner/provisioner_controller.go

```diff
@@ -429,6 +429,7 @@ func (r *ReconcileProvisioner) reconcileSecret(namespace string, secretName stri
 			return err
 		}
 	} else {
+		log.Info("Secret exist in the target cluster. Updating")
 		err = targetClient.Update(ctx, targetSFClusterSecret)
```

Comment:
If `targetClient.Update` creates the secret if its not present, then I feel we can remove the `targetClient.Get` and `targetClient.Create` flow.  🤔

Reply:
I tested the above scenario. Looks like  `targetClient.Update` results in error if secret is not present.